packages/p2p/stores/buy-sell-store.js

```diff
@@ -0,0 +1,360 @@
+    onCancelClick() {
+        this.setShouldShowPopup(false);
+    }
+
+    @action.bound
+    onChangeTableType(event) {
+        this.setTableType(event.target.value);
+    }
+
+    @action.bound
+    onConfirmClick(order_info) {
+        this.general_store.props.setOrderId(order_info.id);
+        this.general_store.redirectTo('orders', { nav: { location: 'buy_sell' } });
+    }
+
+    @action.bound
+    setApiErrorMessage(api_error_message) {
+        this.api_error_message = api_error_message;
+    }
+
+    @action.bound
+    setContactInfo(contact_info) {
+        this.contact_info = contact_info;
+    }
+
+    @action.bound
+    setErrorMessage(error_message) {
+        this.error_message = error_message;
+    }
+
+    @action.bound
+    setFormProps(props) {
+        this.form_props = props;
+    }
+
+    @action.bound
+    setHasMoreItemsToLoad(has_more_items_to_load) {
+        this.has_more_items_to_load = has_more_items_to_load;
+    }
+
+    @action.bound
+    setIsLoading(is_loading) {
+        this.is_loading = is_loading;
+    }
+
+    @action.bound
+    setIsSubmitDisabled(is_submit_disabled) {
+        this.is_submit_disabled = is_submit_disabled;
+    }
+
+    @action.bound
+    setItemOffset(item_offset) {
+        this.item_offset = item_offset;
+    }
+
+    @action.bound
+    setItems(items) {
+        this.items = items;
+    }
+
+    @action.bound
+    setPaymentInfo(payment_info) {
+        this.payment_info = payment_info;
+    }
+
+    @action.bound
+    setInitialReceiveAmount() {
+        this.receive_amount = getRoundedNumber(
+            this.advert.min_order_amount_limit * this.advert.price,
+            this.advert.local_currency
+        );
+    }
+
+    @action.bound
+    setReceiveAmount(receive_amount) {
+        this.receive_amount = receive_amount;
+    }
+
+    @action.bound
+    setSelectedAdState(selected_ad_state) {
+        this.selected_ad_state = selected_ad_state;
+    }
+
+    @action.bound
+    setShouldShowPopup(should_show_popup) {
+        this.should_show_popup = should_show_popup;
+    }
+
+    @action.bound
+    setShouldShowVerification(should_show_verification) {
+        this.should_show_verification = should_show_verification;
+    }
+
+    @action.bound
+    setShowAdvertiserPage(show_advertiser_page) {
+        this.show_advertiser_page = show_advertiser_page;
+    }
+
+    @action.bound
+    setSubmitFormState(submitForm) {
+        this.submitForm = submitForm;
+    }
+
+    @action.bound
+    setTableType(table_type) {
+        this.table_type = table_type;
+    }
+
+    @action.bound
+    setSelectedAdvert(selected_advert) {
+        if (!this.general_store.is_advertiser) {
+            this.setShouldShowVerification(true);
+        } else {
+            this.setSelectedAdState(selected_advert);
+            this.setShouldShowPopup(true);
+        }
+    }
+
+    @action.bound
+    setSubmitForm(submitFormFn) {
+        this.setSubmitFormState(submitFormFn);
+    }
```

Comment:
- Could be renamed for clarity, e.g. `setSubmitFormFn`
- Could we get rid of the `setSubmitFormState` function altogether and assign `this.submitForm(Fn)` right away?